### PR TITLE
Added throughput sending feature into monitoring

### PR DESCRIPTION
**Pull request by jpalac**

### CODE REVIEW

**Review by johnsimons — APPROVED** [review]
LGTM

File: src/ServiceControl.Monitoring/HostApplicationBuilderExtensions.cs [hunk 1]
```diff
@@ -87,6 +88,21 @@ static void ConfigureEndpoint(EndpointConfiguration config, Func<ICriticalErrorC
 
         transportCustomization.CustomizeMonitoringEndpoint(config, transportSettings);
 
+        var serviceControlThroughputDataQueue = settings.ServiceControlThroughputDataQueue;
+        if (!string.IsNullOrWhiteSpace(serviceControlThroughputDataQueue))
+        {
+            if (serviceControlThroughputDataQueue.IndexOf("@") >= 0)
```
Comment (johnsimons) [inline src/ServiceControl.Monitoring/HostApplicationBuilderExtensions.cs:94]: This is my lack of knowledge, but can we rely on all transport to use `@`?

File: src/ServiceControl.Monitoring/Infrastructure/ReportThroughputHostedService.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    class ReportThroughputHostedService(ILogger<ReportThroughputHostedService> logger, IMessageSession session, IEndpointMetricsApi endpointMetricsApi, TimeProvider timeProvider) : BackgroundService
+    {
+        protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+        {
+            logger.LogInformation($"Starting {nameof(ReportThroughputHostedService)}");
+
+            try
+            {
+                //await Task.Delay(DelayStart, cancellationToken);
```
Comment (johnsimons) [inline src/ServiceControl.Monitoring/Infrastructure/ReportThroughputHostedService.cs:20]: why was this removed?

File: src/ServiceControl.Monitoring/Infrastructure/ReportThroughputHostedService.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                {
+                    var average = endpointData[i].Metrics["Throughput"]?.Average ?? 0;
+                    throughputData.EndpointThroughputData[i] = new EndpointThroughputData { Name = endpointData[i].Name, Throughput = Convert.ToInt64(average * throughputMinutes * 60) };
+                }
+
+                await session.Send(throughputData, cancellationToken);
+            }
+        }
+
+        static int throughputMinutes = 5;
```
Comment (johnsimons) [inline src/ServiceControl.Monitoring/Infrastructure/ReportThroughputHostedService.cs:65]: Could use a better name, eg `ReportSendingIntervalInMinutes`

**Review by johnsimons — APPROVED** [review]
Added a few more comments, but overall 👍

File: src/ServiceControl.Monitoring/Infrastructure/ReportThroughputHostedService.cs (hunk 2 above)
Comment (jpalac) [reply src/ServiceControl.Monitoring/Infrastructure/ReportThroughputHostedService.cs:20]: that can be removed altogether - left over from copy/paste

File: src/ServiceControl.Monitoring/HostApplicationBuilderExtensions.cs (hunk 1 above)
Comment (jpalac) [reply src/ServiceControl.Monitoring/HostApplicationBuilderExtensions.cs:94]: We set the queue name, so it's only if the user changes it.
It was code copied from here: https://github.com/Particular/ServiceControl/blob/monitoring-throughput-feature/src/ServiceControl.Audit/Infrastructure/NServiceBusFactory.cs#L36

**Review by abparticular — APPROVED** [review]
